backend/src/modules/users/users.controller.ts

```diff
@@ -1,4 +1,76 @@
-import { Controller } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Param,
+  Body,
+  BadRequestException,
+} from '@nestjs/common';
+import {
+  createUserSchema,
+  updateUserSchema,
+  loginUserSchema,
+  idSchema,
+} from './user.zod';
+import { UsersService } from './users.service';
 
 @Controller('users')
-export class UsersController {}
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Get()
+  async findAll() {
+    return this.usersService.findAll();
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id: string) {
+    const result = idSchema.safeParse(id);
+    if (!result.success) {
+      throw new BadRequestException('Invalid user id');
+    }
+    return this.usersService.findOne(id);
```

Comment:
I encourage adding some spacing between the conditions for better readability (both here and in the surrounding code). 🫁 

```suggestion
    if (!result.success) {
      throw new BadRequestException('Invalid user id');
    }
    
    return this.usersService.findOne(id);
```